app/controllers/Application.scala

```diff
@@ -96,16 +96,18 @@ class Application(
     )).withSettingsSurrogateKey
   }
 
-  def contributionsLanding(countryCode: String): Action[AnyContent] = CachedAction() { implicit request =>
+  def oldOrNewContributionsLanding(countryCode: String): Action[AnyContent] = maybeAuthenticatedAction().async { implicit request =>
+    type Attempt[A] = EitherT[Future, String, A]
     implicit val settings: Settings = settingsProvider.settings()
-    Ok(views.html.main(
-      title = "Support the Guardian | Make a Contribution",
-      description = stringsConfig.contributionsLandingDescription,
-      mainId = s"contributions-landing-page-$countryCode",
-      mainJsBundle = "contributionsLandingPage.js",
-      mainStyleBundle = "contributionsLandingPageStyles.css",
-      scripts = views.html.addToWindow("paymentApiPayPalEndpoint", paymentAPIService.payPalCreatePaymentEndpoint)
-    )).withSettingsSurrogateKey
+    val experiments = settings.switches.experiments
+    if (experiments.get("newPaymentFlow").exists(_.isParticipating)) {
+      request.user.traverse[Attempt, IdUser](identityService.getUser(_)).fold(
+        _ => Ok(newContributions(countryCode, None)),
+        user => Ok(newContributions(countryCode, user))
```

Comment:
to save duplication we can fold it into an option and then `.map(Ok(newContributions(countryCode, _)))`

Reply:
have added the logging. Wasn't quite sure what the other suggestion was, could you write it out?

Reply:
I'm not totally sure whether there's an EitherT needs unwrapping somewhere, but something like this
```suggestion
      val maybeUser = request.user.traverse[Attempt, IdUser](identityService.getUser(_)).leftMap(
          logger.error(s"Error fetching user from identity: $err")
      ).toOption
      Ok(newContributions(countryCode, maybeUser).withSettingsSurrogateKey.withServersideAbTestCookie
```